pkg/agent/openflow/cookie/allocator_test.go

```diff
@@ -59,15 +66,16 @@ func TestConcurrentAllocate(t *testing.T) {
 			assert.Equal(t, round, id.Round(), id.String())
 			assert.Equal(t, seq[i].String(), id.Category().String(), id.String())
 		}
-
 	}
 
 	var wg sync.WaitGroup
 	for i := 0; i < concurrentNum; i++ {
 		wg.Add(1)
 		go func() {
 			defer wg.Done()
-			eachGoroutine()
+			// Each goroutine gets its own rand.Rand instance with unique seeds for independent shuffles
+			goroutineRandSrc := rand.New(rand.NewPCG(mainRand.Uint64(), mainRand.Uint64()))
```

Comment:
`mainRand` should not be safe to use by concurrent goroutines according to the documentation: https://pkg.go.dev/math/rand/v2#Source